### PR TITLE
Improve joystick, and launch it alongside follower

### PR DESCRIPTION
**Pull request by rohbotics**
Launching the joystick alongside the follower is often useful

The joystick now behaves more like the one in ROS1

### CODE REVIEW

File: joy/joy_node.cpp [hunk 1]
```diff
@@ -150,10 +150,20 @@ int main(int argc, char * argv[])
...
-    cmd_vel_pub->publish(cmd_vel_msg);
+    bool deadman = msg->buttons[4] != 0;
+    bool sent_zero = false;
+    if (deadman) {
+        cmd_vel_msg->linear.x = deadman ? -msg->axes[1] * g_scale_linear : 0;
+        cmd_vel_msg->angular.z = deadman ? -msg->axes[0] * g_scale_angular : 0;
+        cmd_vel_pub->publish(cmd_vel_msg);
+        sent_zero = false;
+    }
+    else if (sent_zero) {
```
Comment (gerkey) [inline joy/joy_node.cpp]: Should that be `else if (!sent_zero)` ?

File: turtlebot2_drivers/launch/turtlebot_follow.py [hunk 2]
```diff
@@ -31,3 +31,8 @@ def launch(launch_descriptor, argv):
         name='follower',
         exit_handler=restart_exit_handler,
     )
+    ld.add_process(
+        cmd=['joy_node'],
+        name='joy',
+        exit_handler=restart_exit_handler,
```
Comment (gerkey) [inline turtlebot2_drivers/launch/turtlebot_follow.py]: @dhood identified a problem with the `restart_exit_handler` in #2. Can you try some other exit handlers to see what works well, then propagate the change to the other launch files?

File: joy/joy_node.cpp [hunk 3]
```diff
@@ -150,10 +150,20 @@ int main(int argc, char * argv[])
       else
         break; // if we timed out, let ROS spin to do its stuff
     }
-    bool deadman = msg->buttons[0] != 0;
-    cmd_vel_msg->linear.x = deadman ? -msg->axes[1] * g_scale_linear : 0;
-    cmd_vel_msg->angular.z = deadman ? -msg->axes[0] * g_scale_angular : 0;
-    cmd_vel_pub->publish(cmd_vel_msg);
+    bool deadman = msg->buttons[4] != 0;
+    bool sent_zero = false;
```
Comment (gerkey) [inline joy/joy_node.cpp]: I think that you want `sent_zero` to be declared and initialized before coming into the main `while (rclcpp::ok())` loop (if I understand your intent correctly).

File: joy/joy_node.cpp [hunk 4]
```diff
@@ -150,10 +150,20 @@ int main(int argc, char * argv[])
...
+    if (deadman) {
+        cmd_vel_msg->linear.x = deadman ? -msg->axes[1] * g_scale_linear : 0;
+        cmd_vel_msg->angular.z = deadman ? -msg->axes[0] * g_scale_angular : 0;
```
Comment (rohbotics) [inline joy/joy_node.cpp]: The inline conditional on deadman is redundant

**Comment by rohbotics** [comment]
@gerkey Updated with your notes

**Comment by gerkey** [comment]
Thanks for the fixes. +1

**Comment by rohbotics** [comment]
Can I go ahead and merge this?

**Comment by wjwwood** [comment]
Since it's outside of our normal CI (for now) and @gerkey gave a +1, then yes, I'd say go for it.

**Comment by rohbotics** [comment]
Ok, Thanks

**Pull request merged**